lua/docgen/renderer.lua

```diff
@@ -0,0 +1,174 @@
+local m = {}
+
+local Text = {}
+Text.__index = Text
+
+function Text:new()
+  return setmetatable({
+    state = nil,
+    paragraphes = {},
+    itemizes = {},
+    enumerates = {},
+    order = {}
+  }, Text)
+end
+
+local interpret_state = function(line)
+  if line == '' then
+    return 'newline'
```

Comment:
should make these a table (we can make enums in Lua sorta with metatables)

```
local states = {
  NEWLINE = 'newline',
  ...
}
```

You can make the metatable error on `index` and `newindex`